.github/workflows/ci.yml

```diff
@@ -32,6 +32,34 @@ jobs:
           distribution: 'temurin'
       - run: npm run setup
       - run: npm run test
+  check-docs:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: '16'
+      - name: Install Graphviz
+        uses: tlylt/install-graphviz@v1
+      - name: Install Java
+        uses: actions/setup-java@v3
+        with:
+          java-version: '11'
+          distribution: 'temurin'
+      - run: npm run setup
+      - run: npm run build:web
+      - name: Build MarkBind docs (UG + DG)
+        run: npm run build:docs
+      - name: Check build log for warnings
+        run: |
+          log_file="docs/_markbind/logs/markbind-$(date +'%Y-%m-%d').log"
+          pattern="warn"
```

Comment:
```suggestion
          pattern="^warn:"
```
Nit, if we're specifically looking for warnings called by `logger.warn(...)` in our code, this may be more precise and help avoid false positives (e.g. if the user's filepath contains "warn")?

Reply:
`grep "^[0-9T\-:\.Z]+ - warn:" xxx` doesn't work though, seems to be problematic when used with grep

Reply:
Fixed, this works for me: `"^[0-9T:\.Z\-]\+ - warn:"`

Reply:
> "^[0-9T:\.Z\-]\+ - warn:"

Interesting, the closest that I got was:
![image](https://user-images.githubusercontent.com/41845017/232316294-330a3b5c-9e49-4b74-92ee-c72e51a21b74.png)

The escape of the plus is counter-intuitive, making this regex not work outside of grep. Curious if you know why that is?

Reply:
> The escape of the plus is counter-intuitive, making this regex not work outside of grep. Curious if you know why that is the case?

Hmm it could be more intuitive if we used the `-E` flag for extended regex so the plus doesn't need to be escaped:
`grep "^[0-9T:.Z-]\+ - warn:"` or `grep -E "^[0-9T:.Z-]+ - warn:"`
<img width="831" alt="image" src="https://user-images.githubusercontent.com/61113575/232317702-5c44fcd1-da17-4d81-8b43-587aba065b14.png">

Reply:
Got it thanks